osu.Game.Rulesets.Taiko/UI/DrumTouchInputArea.cs

```diff
@@ -136,12 +168,33 @@ protected override void OnTouchUp(TouchUpEvent e)
             handleUp(e.Touch.Source);
             base.OnTouchUp(e);
         }
-
+        private TaikoAction convertInput(TaikoAction input, TaikoTouchControlType controlType) {
+            switch(controlType) {
+                default:
+                    return input; // Using default controls; Nothing needs to be done
+                case TaikoTouchControlType.DDKK:
+                    switch(input) {
+                        case TaikoAction.LeftRim: return TaikoAction.LeftCentre;
+                        case TaikoAction.LeftCentre: return TaikoAction.RightCentre;
+                        case TaikoAction.RightCentre: return TaikoAction.LeftRim;
+                      //case TaikoAction.RightRim: return TaikoAction.RightRim;
+                    }
+                    return TaikoAction.RightRim;
+                case TaikoTouchControlType.KKDD:
+                    switch(input) {
+                        case TaikoAction.RightRim: return TaikoAction.RightCentre;
+                        case TaikoAction.LeftCentre: return TaikoAction.RightRim;
+                        case TaikoAction.RightCentre: return TaikoAction.LeftCentre;
+                      //case TaikoAction.LeftRim: return TaikoAction.LeftRim;
+                    }
+                    return TaikoAction.LeftRim;
+            }
+        }
```

Comment:
Couldn't this be avoided by retrieving the action of the button the finger fell on in `getTaikoActionFromInput`, rather than how it's hardcoding right now.